lib/cluster/index.ts

```diff
@@ -930,17 +1039,34 @@ class Cluster extends EventEmitter {
       return Promise.resolve(startupNodes);
     }
 
-    return Promise.all(
-      hostnames.map((hostname) => this.dnsLookup(hostname))
-    ).then((ips) => {
-      const hostnameToIP = zipMap(hostnames, ips);
-
-      return startupNodes.map((node) =>
-        hostnameToIP.has(node.host)
-          ? Object.assign({}, node, { host: hostnameToIP.get(node.host) })
-          : node
-      );
-    });
+    if (this.options.useSRVRecord) {
+      return Promise.all(
+        hostnames.map((hostname) => this.resolveSrv(hostname))
+      ).then((ips) => {
+        const hostnameToIP = zipMap(hostnames, ips);
+        let result = [];
+        startupNodes.map((node) => {
+          if (hostnameToIP.has(node.host)) {
+            if (ips[0][0].name) {
+              result.push({ host: ips[0][0].name, port: ips[0][0].port });
```

Comment:
Looks like `hostnameToIP` is not actually used? I suppose it should be something like `hostnameToIP.get(node.host)` instead of `ips[0]`?

Also a nit: I think `.filter().map()` should make the code more clear.